### PR TITLE
Manage pending changes to RW accounts

### PR DESCRIPTION
**Pull request by andre8244**
Perform changes to RoadWarrior accounts without restarting the service immediately

https://github.com/NethServer/dev/issues/6177

### CODE REVIEW

**Comment by andre8244** [comment]
Release after https://github.com/NethServer/nethserver-openvpn/pull/68

File: api/openvpn-rw/read [hunk 1]
```diff
@@ -259,6 +259,14 @@ iOS: https://itunes.apple.com/it/app/openvpn-connect/id590379981
...
         error();
     }
+} elsif ($cmd eq 'pending-changes') {
+    # check for pending changes that need a service restart
+    my $pendingChanges = 0;
+
+    if (-e '/var/spool/openvpn/road-warrior-lock') {
+        $pendingChanges = 1;
+    }
+    $ret->{'pendingChanges'} = $pendingChanges;
```
Comment (edospadoni) [inline api/openvpn-rw/read]: We can improve with this:
```perl
$ret->{'pendingChanges'} = (-e '/var/spool/openvpn/road-warrior-lock') ? 1 : 0;
```

@gsanchietti is the perl magician, maybe he can confirm

File: ui/src/views/OpenVPNRW.vue [hunk 2]
```diff
@@ -22,6 +22,51 @@
...
+    >
+      <button
+        class="btn btn-primary pull-right"
+        data-toggle="modal"
+        data-target="#applyPendingChangesModal"
+      >
+        {{$t('openvpn_rw.apply_changes')}}
+      </button>
+      <span :class="['pficon', 'pficon-warning-triangle-o']"></span>
+      <strong>{{$t('warning')}}.</strong>
```
Comment (edospadoni) [inline ui/src/views/OpenVPNRW.vue]: It's a matter of taste, but we always use `:` after `Warning`. We change it to:
```js
<strong>{{$t('warning')}}:</strong>
```

File: ui/src/views/OpenVPNRW.vue [hunk 3]
```diff
@@ -22,6 +22,51 @@
...
+      <div class="modal-dialog">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h4 class="modal-title">{{$t('openvpn_rw.apply_changes')}}</h4>
+          </div>
+          <form class="form-horizontal" v-on:submit.prevent="applyPendingChanges()">
+            <div class="modal-body">
+              <div class="alert alert-warning alert-dismissable">
+                <span class="pficon pficon-warning-triangle-o"></span>
+                <strong>{{$t('warning')}}.</strong>
```
Comment (edospadoni) [inline ui/src/views/OpenVPNRW.vue]: It's a matter of taste, but we always use `:` after `Warning`. We change it to:
```js
<strong>{{$t('warning')}}:</strong>
```

**Review by edospadoni — CHANGES_REQUESTED** [review]

File: api/openvpn-rw/read (hunk 1 above)
Comment (gsanchietti) [reply api/openvpn-rw/read]: I agree, you can even remove the declaration on line 264: 8 bytes of RAM spared! :D

File: ui/src/views/OpenVPNRW.vue (hunk 2 above)
Comment (andre8244) [reply ui/src/views/OpenVPNRW.vue]: Ok. I actually copied it from the firewall module: https://github.com/NethServer/nethserver-firewall-base/blob/master/ui/src/App.vue#L131

**Review by edospadoni — APPROVED** [review]

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**